Configuration File `.golangci.yml` for `golangci-lint` Runs

Bhargav-InfraCloud opened the pull request:
## Change Overview

Configuration file `.golangci.yml` for `golangci-lint` runs as explained in #1761.

#### Changes
1. Linter: Run 7 default linters (`errcheck`, `gosimple`, `govet`, `ineffassign`, `staticcheck`, `typecheck` & `unused`) from `.golangci.yaml`.
2. Linter: Run linter `gofmt` from `.golangci.yaml`.
3. Linter: Run custom linters (`maligned`, `whitespace`, `gocognit` & `unparam`) from `.golangci.yaml`.
4. Linter: Remove linter `maligned` due to following reasons:
    - Linter `maligned` is archived by owner and deprecated from `golangci-lint`.
    - `maligned` (or it's replacement: govet's fieldalignment) recommends >50 struct's field alignment. Too many changes for much less value.
    - Struct fields might be ordered per their similarities. Rearranging for saving few bytes is not reasonable.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues

- fixes #1761 

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

Run the command:
```
make golint
```

It should pass as it was before. Output similar to:
```
make[1]: Entering directory '/home/bhargav-infracloud/Workspace/go/src/github.com/Bhargav-InfraCloud/kanister'
running CMD in the containerized build environment
Running golangci-lint from config file: .golangci.yml
PASS

PASS

make[1]: Leaving directory '/home/bhargav-infracloud/Workspace/go/src/github.com/Bhargav-InfraCloud/kanister'
```
## To Maintainers
~The `gofmt` and `govet` are ran in `make golint` as well as the following scripts:~
~1. build/test.sh~
~2. examples/aws-rds/postgresql/pgtest/build/test.sh~

~Is there any particular reason for this or can these be eliminated from (or) replaced in scripts?~ This is discussed in the comments.

PrasadG193 commented on .golangci.yml:
```diff
@@ -0,0 +1,31 @@
+run:
+  timeout: 10m                          # golangci-lint run's timeout.
+
+  build-tags:                           # All linters will run with below mentioned build tags.
+    - "integration"
```
@Bhargav-InfraCloud what is the significance of this build tag?

Bhargav-InfraCloud replied:
It enables the linting on files tagged as `integration`.

Currently, the codebase consists of 4 build constraints in multiple files:

1. +build `!ignore_autogenerated`
2. +build `!integration`
3. +build `integration`
4. +build `!unit`

Among these, all but 3rd are negative build constraints and are fine by golangci-lint to do the scan. But 3rd stops golangci-lint from linting on files:

1. pkg/testing/e2e_test.go
2. pkg/testing/integration_register.go
3. pkg/testing/integration_test.go

Though these are all integration tests, minimal formatting linters like `gofmt` & `govet` are still required as per `goreportcard` results found on #1274.

PrasadG193 commented on .golangci.yml:
```diff
@@ -0,0 +1,31 @@
+        - errcheck                      # Errors may be ignored in tests.
+        - unparam                       # Tests might have unused function parameters.
+
+    - text: "`ctx` is unused"           # Context might not be in use in places, but for consistency, we pass it.
+      linters:
+        - unparam
```
We might want to address this. Can be addressed separately

PrasadG193 reviewed (approved):
LGTM

PrasadG193 commented:
> Is there any particular reason for this or can these be eliminated from (or) replaced in scripts?

@Bhargav-InfraCloud `examples/aws-rds/postgresql/pgtest/` is a sample app that users can clone and try. Considering it as a separate project, we have kept its build scripts in its directory.

PrasadG193 commented:
@Bhargav-InfraCloud could you please add attach the output of `make golint` and update the test plan?

Bhargav-InfraCloud commented:
> could you please add attach the output of `make golint` and update the test plan?

@PrasadG193 My bad I missed it. I've updated the desc now and rebased the code with the latest on master. Please check.

Bhargav-InfraCloud commented:
Thanks @PrasadG193 😊
